packages/@react-aria/spinbutton/src/useSpinButton.ts

```diff
@@ -130,9 +130,15 @@ export function useSpinButton(
   // In addition, replace the empty string with the word "Empty" so that iOS VoiceOver does not read "50%" for an empty field.
   textValue = textValue === '' ? stringFormatter.format('Empty') : (textValue || `${value}`).replace('-', '\u2212');
 
+  const animationFrameId = useRef<number>(null);
   useEffect(() => {
     if (isFocused.current) {
-      announce(textValue, 'assertive');
+      if (animationFrameId.current) {
```

Comment:
can we just clear it before the new announce? or does that make a constant stream of weird noises? we're essentially debouncing the announcement?

if we must have the delay, then the raf also needs to be canceled and cleaned up on unmount

Reply:
It seems to work with just `clearAnnouncer('assertive');`. Thanks!